.pipelines/templates/template-push-images-to-acr.yml

```diff
@@ -1,13 +1,21 @@
 parameters:
   rpImageACR: ""
+  acrCredentialsJSON: ""
 steps:
   - script: |
       set -e
       #trap 'set +e; for c in $(docker ps -aq); do docker rm -f $c; done; docker image prune -af ; rm -rf ~/.docker/config.json; rm -rf /run/user/$(id -u $USERNAME)/containers/auth.json' EXIT
 
       export RP_IMAGE_ACR=${{ parameters.rpImageACR }}
 
-      az acr login --name "$RP_IMAGE_ACR"
+      if [ -z ${{ parameters.acrCredentialsJSON }} ]; then
```

Comment:
How does the pipeline authenticate to ACR in this case where the parameter hasn't been provided?

Reply:
If you look back a few e2e steps the pipeline first signs into azure with `az login`.  When e2e is running in the MSIT tenant `az acr login` works because the ACR exists within the MSIT tenant.  Essentially If the user/sp in the original `az login` has permissions to be issued an access token to the acr `az acr login` works.

When e2e runs in the new tenant `az acr login` no longer works because the original `az login` only has permissions o the new tenant, it can't authenticate with ACR through `az acr login`.  We also cannot use cross tenant AAD App/SPs to authenticate with ACR so instead new e2e uses a pre-generated acr token stored in a keyvault and passed to this pipeline in the `parameters.acrCredentialsJSON`